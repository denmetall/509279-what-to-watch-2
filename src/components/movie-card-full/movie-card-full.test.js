import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardFull from "./movie-card-full.jsx";
import filmsMock from '../../mocks/films';

it(`Components MovieCardFull renders correctly`, () => {
  const tree = renderer
    .create(<MovieCardFull film={filmsMock[0]}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
