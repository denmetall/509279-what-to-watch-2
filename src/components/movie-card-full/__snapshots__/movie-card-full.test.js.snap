// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components MovieCardFull renders correctly 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__hero"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link undefined"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Drive
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Triller
          </span>
          <span
            className="movie-card__year"
          >
            2011
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          <a
            className="btn movie-card__button"
            href="add-review.html"
          >
            Add review
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie-card__wrap movie-card__translate-top"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster movie-card__poster--big"
      >
        <img
          alt="The Grand Budapest Hotel poster"
          height="327"
          src="img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <nav
          className="movie-nav movie-card__nav"
        >
          <ul
            className="movie-nav__list"
          >
            <li
              className="movie-nav__item"
            >
              <a
                className="movie-nav__link"
                href="#"
              >
                Overview
              </a>
            </li>
            <li
              className="movie-nav__item movie-nav__item--active"
            >
              <a
                className="movie-nav__link"
                href="#"
              >
                Details
              </a>
            </li>
            <li
              className="movie-nav__item"
            >
              <a
                className="movie-nav__link"
                href="#"
              >
                Reviews
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="movie-card__text movie-card__row"
        >
          <div
            className="movie-card__text-col"
          >
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Director
              </strong>
              <span
                className="movie-card__details-value"
              >
                Wes Andreson
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Starring
              </strong>
              <span
                className="movie-card__details-value"
              >
                Bill Murray, 
                <br />
                Edward Norton, 
                <br />
                Jude Law, 
                <br />
                Willem Dafoe, 
                <br />
                Saoirse Ronan, 
                <br />
                Tony Revoloru, 
                <br />
                Tilda Swinton, 
                <br />
                Tom Wilkinson, 
                <br />
                Owen Wilkinson, 
                <br />
                Adrien Brody, 
                <br />
                Ralph Fiennes, 
                <br />
                Jeff Goldblum
              </span>
            </p>
          </div>
          <div
            className="movie-card__text-col"
          >
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Run Time
              </strong>
              <span
                className="movie-card__details-value"
              >
                1h 39m
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Genre
              </strong>
              <span
                className="movie-card__details-value"
              >
                Comedy
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Released
              </strong>
              <span
                className="movie-card__details-value"
              >
                2014
              </span>
            </p>
          </div>
        </div>
        <div
          className="movie-rating"
        >
          <div
            className="movie-rating__score"
          >
            8,9
          </div>
          <p
            className="movie-rating__meta"
          >
            <span
              className="movie-rating__level"
            >
              Very good
            </span>
            <span
              className="movie-rating__count"
            >
              240 ratings
            </span>
          </p>
        </div>
        <div
          className="movie-card__text"
        >
          <p>
            In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.
          </p>
          <p>
            Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.
          </p>
          <p
            className="movie-card__director"
          >
            <strong>
              Director: Wes Andreson
            </strong>
          </p>
          <p
            className="movie-card__starring"
          >
            <strong>
              Starring: Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other
            </strong>
          </p>
        </div>
        <div
          className="movie-card__reviews movie-card__row"
        >
          <div
            className="movie-card__reviews-col"
          >
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the directors funniest and most exquisitely designed movies in years.
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Kate Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    December 24, 2016
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                8,9
              </div>
            </div>
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the directors funniest and most exquisitely designed movies in years.
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Kate Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    December 24, 2016
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                8,9
              </div>
            </div>
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the directors funniest and most exquisitely designed movies in years.
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Kate Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    December 24, 2016
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                8,9
              </div>
            </div>
          </div>
          <div
            className="movie-card__reviews-col"
          >
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the directors funniest and most exquisitely designed movies in years.
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Kate Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    December 24, 2016
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                8,9
              </div>
            </div>
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the directors funniest and most exquisitely designed movies in years.
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Kate Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    December 24, 2016
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                8,9
              </div>
            </div>
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the directors funniest and most exquisitely designed movies in years.
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Kate Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    December 24, 2016
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                8,9
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
